### ISSUE TITLE
MailVanishedEvent returns sequence ID not UID

### ISSUE THREAD

**Issue by jstride** [Enough-Software/enough_mail#94]
The documentation states this should return a UID  appears to return the sequence ID.

**Comment by robert-virkus** [comment]
To find out the cause, when do you receive this event, is that after selecting a mailbox or just during normal operation after messages have been removed remotely?

**Comment by jstride** [comment]
The later when polling and events are deleted remotely.

**Comment by robert-virkus** [comment]
Thanks, looking deeper this seems to be a documentation problem, as expunge events are mapped to vanished events, too. And expunge events always use sequence IDs. You can check the `MailVanishedEvent.sequence` directly, if it is a UID sequence:
```dart
if (vanishedEvent.sequence.isUidSequence ?? false) {
   // ok, these are UIDs
} else {
  // normal sequence IDs
}
```
Note the null check/false fallback above, with the next release this check will not be required anymore.

**Comment by robert-virkus** [comment]
fixed in v0.0.34 release
